sdks/java/extensions/sql/jdbc/src/test/java/org/apache/beam/sdk/extensions/sql/jdbc/BeamSqlLineIT.java

```diff
@@ -0,0 +1,302 @@
+            + "           ride_status VARCHAR,\n"
+            + "           passenger_count TINYINT>)\n"
+            + "       TYPE pubsub \n"
+            + "       LOCATION '%s'\n"
+            + "       TBLPROPERTIES '{\"timestampAttributeKey\": \"ts\"}';";
+
+    readFromPubsub =
+        "SELECT event_timestamp, taxi_rides.payload.ride_status, taxi_rides.payload.latitude, "
+            + "taxi_rides.payload.longitude from taxi_rides LIMIT 3;";
+
+    filterForSouthManhattan =
+        "SELECT event_timestamp, taxi_rides.payload.ride_status, \n"
+            + "taxi_rides.payload.latitude, taxi_rides.payload.longitude from taxi_rides\n"
+            + "       WHERE taxi_rides.payload.longitude > -74.747\n"
+            + "         AND taxi_rides.payload.longitude < -73.969\n"
+            + "         AND taxi_rides.payload.latitude > 40.699\n"
+            + "         AND taxi_rides.payload.latitude < 40.720 LIMIT 2;";
+
+    fixedWindowQuery =
+        "WITH geo_cells AS (\n"
+            + "         SELECT FLOOR(taxi_rides.payload.latitude / 0.05) * 0.05 AS reduced_lat,\n"
+            + "                FLOOR(taxi_rides.payload.longitude / 0.05) * 0.05 AS reduced_lon,\n"
+            + "                taxi_rides.event_timestamp\n"
+            + "         FROM taxi_rides)\n"
+            + "       SELECT COUNT(*) as num_events,\n"
+            + "              geo_cells.reduced_lat,\n"
+            + "              geo_cells.reduced_lon, \n"
+            + "              TUMBLE_START(geo_cells.event_timestamp, INTERVAL '1' SECOND)\n"
+            + "       FROM geo_cells    \n"
+            + "       GROUP BY geo_cells.reduced_lat,\n"
+            + "                geo_cells.reduced_lon,\n"
+            + "                TUMBLE(geo_cells.event_timestamp, INTERVAL '1' SECOND)\n"
+            + "       LIMIT 2;";
+
+    slidingWindowQuery =
+        "SELECT COUNT(*) AS num_events,\n"
+            + "              SUM(taxi_rides.payload.meter_increment) as revenue,\n"
+            + "              HOP_END(\n"
+            + "                 taxi_rides.event_timestamp, \n"
+            + "                 INTERVAL '1' SECOND, \n"
+            + "                 INTERVAL '2' SECOND) as minute_end\n"
+            + "              FROM taxi_rides\n"
+            + "              GROUP BY HOP(\n"
+            + "                 taxi_rides.event_timestamp,\n"
+            + "                 INTERVAL '1' SECOND, \n"
+            + "                 INTERVAL '2' SECOND) LIMIT 2";
+
+    constructor =
+        new PubsubMessageJSONStringConstructor(
+            "ride_id",
+            "point_idx",
+            "latitude",
+            "longitude",
+            "meter_reading",
+            "meter_increment",
+            "ride_status",
+            "passenger_count");
+  }
+
+  @Test
+  public void testSelectFromPubsub() throws Exception {
+    Future<List<List<String>>> expectedResult =
+        pool.submit(
```

Comment:
This is going to make the thing you are trying to test really hard to debug when the tests fail. Can you put the pubsub generation code in the thread instead?